src/store.ts

```diff
@@ -413,32 +409,115 @@ export default function Store(
-        ) {
-            this.SDKConfig.flags[Constants.FeatureFlags.OfflineStorage] = 0;
+export interface IFeatureFlags {
+    reportBatching?: string;
+    eventBatchingIntervalMillis?: string;
+    offlineStorage?: string;
+    directURLRouting?: boolean;
+}
+
+export function processFlags(
+    config: SDKInitConfig,
+    SDKConfig: SDKConfig
+): IFeatureFlags {
+    const flags: IFeatureFlags = {};
+    const {
+        ReportBatching,
+        EventBatchingIntervalMillis,
+        OfflineStorage,
+        DirectUrlRouting,
+    } = Constants.FeatureFlags;
+
+    if (!config.flags) {
+        return {};
+    }
+
+    // Passed in config flags take priority over defaults
+    flags[ReportBatching] = config.flags[ReportBatching] || false;
+    flags[EventBatchingIntervalMillis] =
+        config.flags[EventBatchingIntervalMillis] ||
+        Constants.DefaultConfig.uploadInterval;
+    flags[OfflineStorage] = config.flags[OfflineStorage] || '0';
+    flags[DirectUrlRouting] = config.flags[DirectUrlRouting] === 'True';
+
+    return flags;
+}
+
+export function processEndpoints(
+    config: SDKInitConfig,
+    flags: IFeatureFlags,
+    apiKey: string
+): Dictionary {
+    // an API key is not present in a webview only mode. In this case, no endpoints are needed
+    if (!apiKey) {
+        return {};
+    }
+
+    // Set default endpoints
+    let endpoints: Dictionary = JSON.parse(
+        JSON.stringify(Constants.DefaultUrls)
+    );
+
+    // When direct URL routing is false, update endpoints based custom urls
+    // passed to the config
+    if (!flags.directURLRouting) {
+        return processNonDirectEndpoints(endpoints, config);
+    } else {
+        return processDirectUrlEndpoints(endpoints, config, apiKey);
+    }
+}
+
+function processNonDirectEndpoints(
+    endpoints: Dictionary,
+    config: SDKInitConfig
+): Dictionary {
+    for (let endpoint in endpoints) {
+        if (config.hasOwnProperty(endpoint)) {
+            endpoints[endpoint] = config[endpoint];
         }
     }
+    return endpoints;
 }
+
+function processDirectUrlEndpoints(
+    endpoints: Dictionary,
+    config: SDKInitConfig,
+    apiKey: string
+): Dictionary {
+    // When Direct URL Routing is true, we create a new set of endpoints that
+    // include the silo in the urls.  mParticle API keys are prefixed with the
+    // silo and a hyphen (ex. "us1-", "us2-", "eu1-").  us1 was the first silo,
+    // and before other silos existed, there were no prefixes and all apiKeys
+    // were us1. As such, if we split on a '-' and the resulting array length
+    // is 1, then it is an older APIkey that should route to us1.
+    const splitKey: Array<string> = apiKey.split('-');
+    let routingPrefix: string;
+
+    // when splitKey.length is greater than 1, then splitKey[0] will be
+    // us1, us2, eu1, au1, or st1, etc as new silos are added
+    routingPrefix = splitKey.length <= 1 ? 'us1' : splitKey[0];
+
+    for (let endpointKey in endpoints) {
```

Comment:
As mentioned in my previous comment, if you don't mutate the endpoints within the process method, you can simplify this method and move this logic outside of the process method, which might make things more readable.